### ISSUE TITLE
Stack state is persisted across restarts

### ISSUE THREAD

**Issue by endorama** [elastic/elastic-package#507]
I'm using `elastic-package stack up -v` to boot the stack for testing purposes.

While working with integrations I noticed that the integration I've been working on was already installed across stack restarts.

My workflow is running (in shell A):
```
integrations $ elastic-package stack up -v
```
Then develop in shell B running:
```
integrations/package/gcp $ elastic-package build
integrations/package/gcp $ elastic-package stack up --services package-registry -d
```

To restart the stack I'm terminating the `up` process in shell a with `CTRL-c` the starting it again.

Once I install the `gcp` package though, it's reported as installed across restarts. This suggest the cluster state is being persisted.

This behaviour is at best counter intuitive, and I've not been able to understand which docker feature is producing it. The services do not have a volume attached for persistence, but running `elastic-package stack down` destroy the state, so the hypothesis is that it happens because volumes are destroyed by that command; see https://github.com/elastic/elastic-package/blob/9b5a0fa512e3de83f47545ae1b879d1f21720a2a/internal/stack/compose.go#L100-L104

By not using `-d` this behaviour is more visible because to stop containers is enough to `CTRL-c` the `up` process.  
When using `-d` the easiest way to stop the stack is `elastic-package stack down`; using this workflow this behaviour is not observed.

OS: `elementary OS 5.1.7 Hera` (based on `Ubuntu 18.04.4 LTS)`
docker version:
```
Client: Docker Engine - Community
 Version:           20.10.8
 API version:       1.41
 Go version:        go1.16.6
 Git commit:        3967b7d
 Built:             Fri Jul 30 19:54:08 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.8
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.6
  Git commit:       75249d8
  Built:            Fri Jul 30 19:52:16 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.9
  GitCommit:        e25210fe30a0a703442421b0f60afac609f950a3
 runc:
  Version:          1.0.1
  GitCommit:        v1.0.1-0-g4144b63
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```
docker-compose version:
```
docker-compose version 1.28.6, build unknown
docker-py version: 4.4.4
CPython version: 3.8.9
OpenSSL version: OpenSSL 1.1.1k  25 Mar 2021
```

**Comment by endorama** [comment]
> The services do not have a volume attached for persistence, but running elastic-package stack down destroy the state, so the hypothesis is that it happens because volumes are destroyed by that command;

In order to verify this hypothesis I'm changing the code to not use `--volumes` extra args and check the resulting behaviour.

**Comment by endorama** [comment]
The hypothesis is not confirmed, as the same behaviour is observed when `docker-compose down` is not run with `--volumes`.

But testing it I found the root case: when you stop a docker container the container state is preserved. In order to erase that state `docker rm` should be run. `docker-compose down` is removing containers and that is why `elastic-package down` has the "expected behaviour".

In the end this is me being confused by a docker feature :sweat: 

From a UX point of view though I think there may be some improvements we could make to the tool to make this behaviour more visible and even use it at our own advantage (in https://github.com/elastic/elastic-package/issues/498 I was asking for persistent storage to be added).

**Comment by mtojek** [comment]
Thank you for raising both issues (https://github.com/elastic/elastic-package/issues/507 and https://github.com/elastic/elastic-package/issues/498). Apparently we have field to improve, even though it's just UX :)

**Comment by endorama** [comment]
I spent some time thinking out to improve the current situation and I would propose to introduce these changes:
- add `start` and `stop` commands, to mimic `docker` and `docker-compose`
- remove `-d` and always run the stack in daemon mode; by adding `start`, `stop` and this change the behaviour may be easily explained, plus it also would adhere to what docker does (so is easier to not be confused by different expectations)
	Eventually is possible to mimic the current behaviour by showing the logs after start (so like `docker-compose up; docker-compose logs`); this way CTRL-c would not stop the containers but only log tailing
- enhance the doc to describe this use case, helped by the new commands and the removal of foreground `stack up`

/cc @jsoriano

(separately we can discuss if #498 still make sense, which I think it does albeit with a slightly different use case)

**Comment by jsoriano** [comment]
Thanks @endorama for your thoughts on this, I think this would be a nice thing to have.

> * add `start` and `stop` commands, to mimic `docker` and `docker-compose`

I think `stop` would make totally sense, to be able to stop the stack in daemon mode without removing it. For `start` I had to think it twice, I guess it would be needed to avoid recreating the containers if something has changed, right?

> * remove `-d` and always run the stack in daemon mode; by adding `start`, `stop` and this change the behaviour may be easily explained, plus it also would adhere to what docker does (so is easier to not be confused by different expectations)
>   Eventually is possible to mimic the current behaviour by showing the logs after start (so like `docker-compose up; docker-compose logs`); this way CTRL-c would not stop the containers but only log tailing

I think I like to have the option of an attached mode as is the current default. But this could be discussed, I would be also ok changing the defaults but still having a flag for the attached mode. We have to take into account that removing flags or changing defaults would be a breaking change, and elastic-package is GA.

> * enhance the doc to describe this use case, helped by the new commands and the removal of foreground `stack up`

+1 about enhancing docs, but even better if we have a developer experience that causes few surprises and needs few explanations :slightly_smiling_face: 

An extra use case that I find could be valuable and I think we don't have and we wouldn't still have with the proposed changes, is to be able to update a stack without losing the data. For that we probably need volumes for the data directories in any case.

**Comment by endorama** [comment]
> For start I had to think it twice, I guess it would be needed to avoid recreating the containers if something has changed, right?

Once you run `stop` how do you start the stack again? `start` command feels the most intuitive. I'm not sure it needs a specific implementation or may just be an alias for `up`.

> I think I like to have the option of an attached mode as is the current default. But this could be discussed, I would be also ok changing the defaults but still having a flag for the attached mode. We have to take into account that removing flags or changing defaults would be a breaking change, and elastic-package is GA.

I think that this change may be postponed if the other 2 points are implemented. So in the docs we always specify `-d` as the preferred way and that would improve UX by making `start`, `stop` and `down` more ergonomic.

> An extra use case that I find could be valuable and I think we don't have and we wouldn't still have with the proposed changes, is to be able to update a stack without losing the data. For that we probably need volumes for the data directories in any case.

I moved this comment to #498 as this is a super valid use case!

**Comment by jsoriano** [comment]
> > For start I had to think it twice, I guess it would be needed to avoid recreating the containers if something has changed, right?
> 
> Once you run `stop` how do you start the stack again? `start` command feels the most intuitive. I'm not sure it needs a specific implementation or may just be an alias for `up`.

Yep, I was thinking on `up` as the single command to start the stack, but there may be differences, `up` replaces things that changed in the docker compose definition, while start would just start the existing containers.
